File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = set(nums)
        max_val = 0
        length = 0
        for i in nums:
            if i-1 not in nums:
                length = 0
                while i+length in nums:
                    length += 1
            max_val = max(max_val,length)
        return max_val

                
                

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> myset = new HashSet<Integer>();
        int result = 0;

        for(int num: nums) {
            myset.add(num);
        }

        for (int num: myset) {
            if (!myset.contains(num-1)) {
                int start  = num;
                int count = 1;
                while (myset.contains(start+1)) {
                    start++;
                    count++;
                }
                result = Math.max(result, count);
            }  
        }

        return result;
     }
}